src/components/Select/Select.tsx

```diff
@@ -0,0 +1,147 @@
+import { Option, ChangeHandler, useSelectEvents } from "./useSelectEvents";
+import { SelectWrapper } from "./SelectWrapper";
+import { List, Text, Box, PropsWithBox } from "..";
+import { helperTextRecipe, InputVariants } from "../BaseInput";
+import { listStyle, listItemStyle, listWrapperRecipe } from "../BaseSelect";
+
+export type SelectProps = PropsWithBox<
+  Omit<
+    InputHTMLAttributes<HTMLSelectElement>,
+    | "color"
+    | "width"
+    | "height"
+    | "size"
+    | "type"
+    | "children"
+    | "onChange"
+    | "value"
+    | "nonce"
+  > & {
+    label?: ReactNode;
+    error?: boolean;
+    helperText?: ReactNode;
+    options: Option[];
+    onChange?: ChangeHandler;
+    value: string | number;
+  }
+> &
+  InputVariants;
+
+const getBoxHeight = (size: SelectProps["size"]) => {
+  switch (size) {
+    case "small":
+      return 7;
+    case "medium":
+      return 8;
+    case "large":
+      return 9;
+  }
+};
+
+export const Select = forwardRef<HTMLSelectElement, SelectProps>(
+  (
+    {
+      size = "medium",
+      disabled = false,
+      className,
+      value,
+      label,
+      id,
+      error = false,
+      helperText,
+      options,
+      onChange,
+      onFocus,
+      onBlur,
+      ...props
+    },
+    ref
+  ) => {
+    const {
+      active,
+      typed,
+      isOpen,
+      getToggleButtonProps,
+      getLabelProps,
+      getMenuProps,
+      highlightedIndex,
+      getItemProps,
+      selectedItem,
+      handlers,
+    } = useSelectEvents(value, options, onChange);
+
+    return (
+      <Box display="flex" flexDirection="column" gap={3}>
+        <SelectWrapper
+          id={id}
+          typed={typed}
+          active={active}
+          disabled={disabled}
+          size={size}
+          label={label}
+          error={error}
+          className={className}
+          getLabelProps={getLabelProps}
+          getToggleButtonProps={getToggleButtonProps}
+        >
+          <Box
+            height={getBoxHeight(size)}
+            onBlur={(event: FocusEvent<HTMLSelectElement, Element>) => {
+              handlers.onBlur();
+              onBlur?.(event);
+            }}
+            onFocus={(event: FocusEvent<HTMLSelectElement, Element>) => {
+              handlers.onFocus();
+              onFocus?.(event);
+            }}
+            {...props}
+            ref={ref}
+          >
+            <Text size={size} variant="body">
+              {selectedItem?.label}
+            </Text>
+          </Box>
+        </SelectWrapper>
+
+        <Box
+          position="relative"
+          display={isOpen ? "block" : "none"}
+          className={listWrapperRecipe({ size })}
+        >
+          <List as="ul" className={listStyle} {...getMenuProps()}>
+            {isOpen &&
+              options?.map((item, index) => (
+                <List.Item
+                  key={`${id}-${item}-${index}`}
+                  className={listItemStyle}
+                  {...getItemProps({
```

Comment:
issue: I don't suggest at all to use that way of delivering nested props, it doesn't scale. We limit uset to certain appearance while it's not always the case, sometimes there is a need to have more customization than simply displaying text.

Everywhere you use getXXXProps we can use either render-props or deliver certain data by context + expose some declarative wrappers for the user